src/ORM/Table.php

```diff
@@ -1728,7 +1728,8 @@ protected function _update($entity, $data)
         }
 
         if (!$entity->has($primaryColumns)) {
-            $message = 'All primary key value(s) are needed for updating';
+            $message = 'All primary key value(s) are needed for updating, ';
+            $message .= get_class($entity) . ' is missing ' . implode(', ', $primaryColumns);
```

Comment:
**Suggestion**
`When updating a ENTITIY the (entire) primary key (KEY/S) is required but a value/values for (MISSING-KEY/S) was missing.`

**3 Examples**
When updating a Post the primary key (id) is required but a value for (id) was missing.

When updating a Comment the entire primary key (id, parent_id) is required but a value for (parent_id) was missing.

When updating a Comment the entire primary key (post_id, id, parent_id) is required but values for (post_id, parent_id) were missing.